util/generate_compilation_db.py

```diff
@@ -18,17 +18,23 @@
 Example:
   util/generate_compilation_db.py --target //sw/... --out compile_commands.json
 
+Tip: If your IDE complains that it cannot find headers, e.g. "gmock/gmock.h", it
+might be telling the truth. Try building the relevant target with Bazel
+(specifying "--config=riscv32" as necessary) and then restart clangd.
+
```

Comment:
This is a bit of a rough edge. If you're going back and forth between e.g. `sram_program.c` and `memory_unittest.cc`, you may have to keep running `bazel build --config=riscv32 //sw/device/examples/sram_program` and `bazel build //sw/device/lib/base:memory_unittest`.

The problem seems to be that Bazel changes the symlinks in `bazel-opentitan/external` depending on the platform of the current build.

```console
dmcardle@multivac01 ~/REPOS/opentitan % ls -l bazel-opentitan/external
total 36
lrwxrwxrwx 1 dmcardle primarygroup 114 Oct 10 18:04 bazel_tools -> /usr/local/google/home/dmcardle/.cache/bazel/_bazel_dmcardle/b5c6bad67a60fe0f0e2dfcf06963df7b/external/bazel_tools
lrwxrwxrwx 1 dmcardle primarygroup 113 Oct 10 18:04 googletest -> /usr/local/google/home/dmcardle/.cache/bazel/_bazel_dmcardle/b5c6bad67a60fe0f0e2dfcf06963df7b/external/googletest
lrwxrwxrwx 1 dmcardle primarygroup 118 Oct 10 18:04 local_config_cc -> /usr/local/google/home/dmcardle/.cache/bazel/_bazel_dmcardle/b5c6bad67a60fe0f0e2dfcf06963df7b/external/local_config_cc
lrwxrwxrwx 1 dmcardle primarygroup 112 Oct 10 18:04 platforms -> /usr/local/google/home/dmcardle/.cache/bazel/_bazel_dmcardle/b5c6bad67a60fe0f0e2dfcf06963df7b/external/platforms
lrwxrwxrwx 1 dmcardle primarygroup 135 Oct 10 18:04 python3_x86_64-unknown-linux-gnu -> /usr/local/google/home/dmcardle/.cache/bazel/_bazel_dmcardle/b5c6bad67a60fe0f0e2dfcf06963df7b/external/python3_x86_64-unknown-linux-gnu
lrwxrwxrwx 1 dmcardle primarygroup 124 Oct 10 18:04 remote_coverage_tools -> /usr/local/google/home/dmcardle/.cache/bazel/_bazel_dmcardle/b5c6bad67a60fe0f0e2dfcf06963df7b/external/remote_coverage_tools
lrwxrwxrwx 1 dmcardle primarygroup 120 Oct 10 18:04 remote_java_tools -> /usr/local/google/home/dmcardle/.cache/bazel/_bazel_dmcardle/b5c6bad67a60fe0f0e2dfcf06963df7b/external/remote_java_tools
lrwxrwxrwx 1 dmcardle primarygroup 126 Oct 10 18:04 remote_java_tools_linux -> /usr/local/google/home/dmcardle/.cache/bazel/_bazel_dmcardle/b5c6bad67a60fe0f0e2dfcf06963df7b/external/remote_java_tools_linux
lrwxrwxrwx 1 dmcardle primarygroup 120 Oct 10 18:04 remotejdk11_linux -> /usr/local/google/home/dmcardle/.cache/bazel/_bazel_dmcardle/b5c6bad67a60fe0f0e2dfcf06963df7b/external/remotejdk11_linux
dmcardle@multivac01 ~/REPOS/opentitan % bazel build --config=riscv32 //sw/device/examples/sram_program
INFO: Invocation ID: 1ed62dd5-0a8a-4e61-9057-6cab8f01fc2b
INFO: Build option --platforms has changed, discarding analysis cache.
INFO: Analyzed target //sw/device/examples/sram_program:sram_program (0 packages loaded, 6804 targets configured).
INFO: Found 1 target...
Target //sw/device/examples/sram_program:sram_program up-to-date (nothing to build)
INFO: Elapsed time: 0.714s, Critical Path: 0.12s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
dmcardle@multivac01 ~/REPOS/opentitan % ls -l bazel-opentitan/external
total 28
lrwxrwxrwx 1 dmcardle primarygroup 114 Oct 10 18:04 bazel_tools -> /usr/local/google/home/dmcardle/.cache/bazel/_bazel_dmcardle/b5c6bad67a60fe0f0e2dfcf06963df7b/external/bazel_tools
lrwxrwxrwx 1 dmcardle primarygroup 106 Oct 10 18:04 crt -> /usr/local/google/home/dmcardle/.cache/bazel/_bazel_dmcardle/b5c6bad67a60fe0f0e2dfcf06963df7b/external/crt
lrwxrwxrwx 1 dmcardle primarygroup 118 Oct 10 18:04 local_config_cc -> /usr/local/google/home/dmcardle/.cache/bazel/_bazel_dmcardle/b5c6bad67a60fe0f0e2dfcf06963df7b/external/local_config_cc
lrwxrwxrwx 1 dmcardle primarygroup 125 Oct 10 18:04 lowrisc_rv32imcb_files -> /usr/local/google/home/dmcardle/.cache/bazel/_bazel_dmcardle/b5c6bad67a60fe0f0e2dfcf06963df7b/external/lowrisc_rv32imcb_files
lrwxrwxrwx 1 dmcardle primarygroup 117 Oct 10 18:04 ot_python_deps -> /usr/local/google/home/dmcardle/.cache/bazel/_bazel_dmcardle/b5c6bad67a60fe0f0e2dfcf06963df7b/external/ot_python_deps
lrwxrwxrwx 1 dmcardle primarygroup 112 Oct 10 18:04 platforms -> /usr/local/google/home/dmcardle/.cache/bazel/_bazel_dmcardle/b5c6bad67a60fe0f0e2dfcf06963df7b/external/platforms
lrwxrwxrwx 1 dmcardle primarygroup 135 Oct 10 18:04 python3_x86_64-unknown-linux-gnu -> /usr/local/google/home/dmcardle/.cache/bazel/_bazel_dmcardle/b5c6bad67a60fe0f0e2dfcf06963df7b/external/python3_x86_64-unknown-linux-gnu
```